clouddriver-google/src/main/groovy/com/netflix/spinnaker/clouddriver/google/deploy/GCEUtil.groovy

```diff
@@ -80,54 +80,33 @@ class GCEUtil {
                           GoogleExecutorTraits executor) {
     task.updateStatus phase, "Looking up image $imageName..."
 
+    def filter = "name eq $imageName"
     def imageProjects = [projectName] + credentials?.imageProjects + baseImageProjects - null
     def sourceImage = null
-    // We want predictable iteration order that matches the order of insertion.
-    def imageProjectToNextPageTokenMap = new LinkedHashMap<String, String>()
 
-    // This will ensure that each image project is queried.
-    imageProjects.each { imageProjectToNextPageTokenMap[it] = null }
-
-    while (!sourceImage && imageProjectToNextPageTokenMap) {
-      def imageListBatch = buildBatchRequest(compute, clouddriverUserAgentApplicationName)
-      def imageListCallback = new JsonBatchCallback<ImageList>() {
-        @Override
-        void onFailure(GoogleJsonError e, HttpHeaders responseHeaders) throws IOException {
-          updateStatusAndThrowNotFoundException("Error locating $imageName in these projects: $imageProjects: $e.message", task, phase)
-        }
-
-        @Override
-        void onSuccess(ImageList imageList, HttpHeaders responseHeaders) throws IOException {
-          // No need to look through these images if the requested image was already found.
-          if (!sourceImage) {
-            sourceImage = imageList.items.find { it.name == imageName }
-          }
-
-          // selfLinks look like this: https://www.googleapis.com/compute/alpha/projects/ubuntu-os-cloud/global/images
-          def selfLinkTokens = imageList.getSelfLink().tokenize("/")
-          def imageProject = selfLinkTokens[selfLinkTokens.size() - 3]
-
-          if (imageList.nextPageToken) {
-            imageProjectToNextPageTokenMap[imageProject] = imageList.nextPageToken
-          } else {
-            imageProjectToNextPageTokenMap.remove(imageProject)
-          }
-        }
+    def imageListBatch = buildBatchRequest(compute, clouddriverUserAgentApplicationName)
+    def imageListCallback = new JsonBatchCallback<ImageList>() {
+      @Override
+      void onFailure(GoogleJsonError e, HttpHeaders responseHeaders) throws IOException {
+        updateStatusAndThrowNotFoundException("Error locating $imageName in these projects: $imageProjects: $e.message", task, phase)
```

Comment:
yea, as I noted offline, I'm ok as is. Just wanted to bring this to your attention.
I was assuming projects was essentially a set and not a list so adding or removing a project (even in another execution) would change the order even if the added/removed project did not contain the image in question. That would situation would be really nasty to debug.